src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxDocumentConverter.cs

```diff
@@ -0,0 +1,252 @@
+
+                case ValueType.DateTimeOffset:
+                    var dateTimeOffset = (DateTimeOffset)value;
+                    var dateTimeOffsetBytes = dateTimeOffset.UtcDateTime.GetDefaultRavenFormat(isUtc: true);
+                    scope.Write(field.Id, dateTimeOffsetBytes, ref entryWriter);
+                    return;
+
+                case ValueType.TimeSpan:
+                    var timeSpan = (TimeSpan)value;
+                    scope.Write(field.Id, timeSpan.ToString("c", CultureInfo.InvariantCulture), ref entryWriter);
+                    return;
+
+                case ValueType.Convertible:
+                    scope.Write(field.Id, ((IConvertible)value).ToString(CultureInfo.InvariantCulture), ref entryWriter);
+                    return;
+
+                case ValueType.Enumerable:
+                    var iterator = (IEnumerable)value;
+                    var enumerableScope = new EnumerableWriterScope(field.Id, ref entryWriter, scope.GetLengthList());
+                    foreach (var item in iterator)
+                    {
+                        InsertRegularField(field, item, indexContext, out _, ref entryWriter, enumerableScope);
+                    }
+                    enumerableScope.Finish(field.Id, ref entryWriter);
+                    return;
+
+                case ValueType.DynamicJsonObject:
+                    HandleObject((BlittableJsonReaderObject)value, field, indexContext, out _, ref entryWriter, scope);
+                    return;
+
+                case ValueType.ConvertToJson:
+                    var val = TypeConverter.ToBlittableSupportedType(value);
+                    if (!(val is DynamicJsonValue json))
+                        InsertRegularField(field, val, indexContext, out _, ref entryWriter, scope, nestedArray);
+                    return;
+
+                case ValueType.BlittableJsonObject:
+                    HandleObject((BlittableJsonReaderObject)value, field, indexContext, out _, ref entryWriter, scope);
+                    return; 
+
+                case ValueType.BoostedValue:
+                case ValueType.Stream:
+                case ValueType.DynamicNull:
+                case ValueType.Null:
+                default:
+                    throw new NotImplementedException();
+            }
+
+            shouldSkip = true;
+        }
+
+        void HandleArray(IEnumerable itemsToIndex, IndexField field, JsonOperationContext indexContext, out bool shouldSkip, ref IndexEntryWriter entryWriter, IWriterScope scope, bool nestedArray = false)
+        {
+            shouldSkip = false;
+            if (nestedArray)
+            {
+                return;
+            }
+
+            foreach (var itemToIndex in itemsToIndex)
+            {
+                InsertRegularField(field, itemToIndex, indexContext, out _, ref entryWriter, scope);
+            }
+        }
+
+        //todo maciej Discuss how we gonna handle nestedArrays. Now I skip them.
```

Comment:
We flatten all nested arrays: `[ [foo], [bar]]` is the same as `[foo, bar]`

Reply:
@maciejaszyk is it handled? I believe it will be the same as we do in Lucene converter

Reply:
1. For enumerable:
Recursively adding all items to index (so I'm flattening it.)
2. For nested BlittableJsonObject/ConvertToJson
Still need to change it.